Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/CosmosServerlessSparkBatchJob.java

```diff
@@ -143,58 +160,191 @@ public String getJobUuid() {
                 });
     }
 
-    @Override
-    public Observable<SparkBatchJobResponsePayload> getStatus() {
+    private Observable<com.microsoft.azure.hdinsight.sdk.rest.azure.serverless.spark.models.SparkBatchJob> getSparkBatchJob() {
         return getAccount().getSparkBatchJobRequest(getJobUuid())
                 .flatMap(sparkBatchJob -> {
-                    if (sparkBatchJob.properties() == null) {
-                        return Observable.error(new IOException("Can't get job status from empty property body."));
+                    if (sparkBatchJob.schedulerState() != null &&
```

Comment:
The status filter can be moved out of `getSparkBatchJob()` method

Reply:
Since `getResponsePayloadWithState` call `getSparkBatchJob`, do you suggest we move the `filter` into `getResponsePayloadWithState`?